### PR TITLE
KRACOEUS-8728

### PR DESCRIPTION
**Pull request by rmancher**
KRACOEUS-8728
Fix to NOT popup budget sync rates dialog when in view mode
Do not display lock message in view mode - both PD and Budget
Also as per PR comments earlier, I have set to establish locking when NOT in view mode. 
I did not see any issue at this point but thought we don't have to when budget is in view only mode.

### CODE REVIEW

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java [hunk 1]
```diff
@@ -80,9 +83,9 @@ public ModelAndView start(@RequestParam("budgetId") Long budgetId, @RequestParam
         ModelAndView result = getModelAndViewService().getModelAndViewWithInit(form, ProposalBudgetConstants.KradConstants.BUDGET_DEFAULT_VIEW);
         form.setViewOnly(inViewMode);
         form.setCanEditView(!inViewMode);
-        if (budgetRatesService.checkActivityTypeChange(form.getBudget().getBudgetRates(), form.getDevelopmentProposal().getActivityTypeCode())) {
+        if (!inViewMode && budgetRatesService.checkActivityTypeChange(form.getBudget().getBudgetRates(), form.getDevelopmentProposal().getActivityTypeCode())) {
```
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: So I wonder if this viewMode idea needs to be more global. What happens if they only have view permission? Or if the budget is complete and therefore read-only?

Comment (rmancher) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: @blackcathacker  good catch. 
Initially I tried to set it up with view.readOnly which should work for all possible options.
I saw some issues/conflict with budget settings and to set budget in view mode when PD is opened in view mode.
Looking into it to see if that will still work as a global solution.

**Comment by rmancher** [comment]
@blackcathacker I have made some changes to invoke confirmation dialog based on authorization.

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java [hunk 2]
```diff
@@ -80,9 +83,14 @@ public ModelAndView start(@RequestParam("budgetId") Long budgetId, @RequestParam
         ModelAndView result = getModelAndViewService().getModelAndViewWithInit(form, ProposalBudgetConstants.KradConstants.BUDGET_DEFAULT_VIEW);
         form.setViewOnly(inViewMode);
         form.setCanEditView(!inViewMode);
-        if (budgetRatesService.checkActivityTypeChange(form.getBudget().getBudgetRates(), form.getDevelopmentProposal().getActivityTypeCode())) {
+    	boolean canModify = false;
+    	if(!inViewMode) {
+    		canModify = getProposalBudgetAuthorizer().canModifyBudget(form.getBudget(), getGlobalVariableService().getUserSession().getPerson());
```
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: So KIM checks are already run, probably multiple times and can be expensive. You should be able to just check for this in the editModes on the form?

Comment (rmancher) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: I don't think KIM checks are run at this point. Edit modes are not available.

Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: Hmm...ok. Thats frustrating but ok at least its only when we open the doc.

**Pull request merged**